lib/Parse/ParseExpr.cpp

```diff
@@ -2905,9 +2913,49 @@ ExprResult Parser::ParseBoundsExpression() {
     SkipUntil(tok::r_paren, StopAtSemi | StopBeforeMatch);
 
   PT.consumeClose();
+
+  if (StartsRelativeBoundsClause(Tok)) {
+    ParseRelativeBoundsClause(Result.get());
+  }
+
   return Result;
 }
 
+bool Parser::ParseRelativeBoundsClause(Expr *Expr) {
+  RelativeBoundsClause *Result;
+  ExprResult Results;
```

Comment:
The Result.get() could fail if the Result is invalid.   You may need to check if the Result is invalid and pass a null pointer into ParseRelativeBoundsClause.   Then in ParseRelativeBoundsClause you can check that the pointer is non-null before trying to attach the bounds expression.